src/app/interfaces/menusInterfaces.ts

```diff
@@ -199,12 +195,7 @@ export function menuAction<T extends Omit<MenuAction, "kind">>(
  * Any Menu Item discriminated union. This is used to store any of the
  *  menu item types together in a collection.
  */
-export type AnyMenuItem =
-  | MenuAction
-  | MenuLink
-  | MenuRoute
-  | MenuModalWithoutAction
-  | MenuModal;
```

Comment:
Hmm  why was `MenuModal` removed from being a valid menu item?

Reply:
There is no current scenario where i need a type that allows both `MenuModalWithoutAction` and `MenuModal`. Often i need a type which combines `MenuLink|MenuRoute|MenuAction` with either `MenuModalWithoutAction` or `MenuModal`. Potentially i need a better naming scheme. `AnyMenuItem (MenuLink|MenuRoute|MenuAction|MenuModal)`/`AnyMenuItemWithoutModalAction(MenuLink|MenuRoute|MenuAction|MenuModalWithoutAction)`? Currently, I use `AnyMenuItem | MenuModalWithoutAction` or `AnyMenuItem | MenuModal` where needed

Reply:
I guess I just see a menu as a collection of menu items... I'd see no reason at all to ever be more specific about the types.

I don't understand the mutual exclusion for the two modal types. Explain to me on chat? Why can't we have a menu that has multiple modals, of both types?